pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -235,21 +236,54 @@ def setSuns(self, Ee, cells=None):
         """
         if cells is None:
             if np.isscalar(Ee):
-                for pvc in self.pvcells:
+                new_pvcells = range(self.numberCells)  # new list of cells
+                old_pvcells = dict.fromkeys(self.pvcells)  # same as set(pvcells)
+                for cell_id, pvcell in enumerate(self.pvcells):
+                    if old_pvcells[pvcell] is None:
+                        new_pvcells[cell_id] = copy(pvcell)
+                        old_pvcells[pvcell] = new_pvcells[cell_id]
+                    else:
+                        new_pvcells[cell_id] = old_pvcells[pvcell]
+                self.pvcells = new_pvcells
+                for pvc in set(self.pvcells):
```

Comment:
actually, I think `set(self.pvcells)` is just `old_pvcells.itervalues()` right? I think that's better, probably faster (by a femto second) since the set is already calculated and it's an iterator